Assets/MixedRealityToolkit/Services/BaseDataProviderAccessCoreSystem.cs

```diff
@@ -45,21 +78,73 @@ public override void Update()
+            // response to user input, such as a button press. Since most code runs on Unity's main thread, the collection
+            // of data providers may change mid-Update, causing an InvalidOperationException to be thrown during enumeration.
+            // To avoid unneccessary error logging, we catch the exception. It is safe to abort the loop in this fashion since
+            // it is extremely unlikely that the data providers being enumerated are the same as when we started enumeration.
+            try
             {
-                provider.Update();
+                foreach (IMixedRealityDataProvider provider in dataProviders)
+                {
+                    try
+                    {
+                        provider.Update();
+                    }
+                    catch (Exception e)
+                    {
+                        Debug.LogError($"{provider.Name}'s Update method threw {e.GetType()}");
+                    }
+                }
             }
+            catch (InvalidOperationException)
+            { }
         }
 
         /// <inheritdoc />
         public override void LateUpdate()
         {
             base.LateUpdate();
 
-            foreach (var provider in dataProviders)
+            // Applications may choose to trigger service changes (i.e. setting MixedRealityToolkit.ActiveProfile) in
+            // response to user input, such as a button press. Since most code runs on Unity's main thread, the collection
+            // of data providers may change mid-LateUpdate, causing an InvalidOperationException to be thrown during enumeration.
+            // To avoid unneccessary error logging, we catch the exception. It is safe to abort the loop in this fashion since
+            // it is extremely unlikely that the data providers being enumerated are the same as when we started enumeration.
+            try
+            {
+                foreach (IMixedRealityDataProvider provider in dataProviders)
+                {
+                    try
+                    {
+                        provider.LateUpdate();
+                    }
+                    catch (Exception e)
+                    {
+                        Debug.LogError($"{provider.Name}'s LateUpdate method threw {e.GetType()}");
+                    }
+                }
+            }
+            catch (InvalidOperationException)
+            { }
+        }
+
+        /// <inheritdoc />
+        public override void Destroy()
+        {
+            foreach (IMixedRealityDataProvider provider in dataProviders)
             {
-                provider.LateUpdate();
+                try
+                {
+                    provider.Destroy();
+                }
+                catch (Exception e)
+                {
+                    Debug.LogError($"{provider.Name}'s Destroy method threw {e.GetType()}");
+                }
             }
+
+            base.Destroy();
```

Comment:
You are not doing the try / catch / ignore InvalidOperationException here, why?

Reply:
An earlier instance of this PR had exceptions around these calls as well. On further analysis, ResetConfiguration() calls Destroy() prior to making any changes to the collection of registered services. This avoids the issue of changes during enumeration.

That said, adding the exception handing code here as a reasonable request from a defensive coding perspective. Should I add it?

Reply:
I think just a comment explaining why the exception is not here but is in other places is enough. Catching silently might hide other issues.